safe-global/web-core
# feat: Add smart contract wallet support when signing txs

## What it solves

Resolves #503 

## How this PR fixes it

Being connected to the safe via a smart contract wallet is handled when signing a transaction.

- Adds a new `txEvent` called `AWAITING_ON_CHAIN_SIGNATURE`

## How to test it

1. Open a Safe
2. Connect to safe 1 via WC safe app on another safe 2
3. Create a new transaction on safe 1
4. Observe a popup via WC on safe 2
5. Confirm the transaction on safe 2
6. Observe a notification on safe 1
7. Observe being navigated to a proposed tx on safe 1
8. Observe the proposed tx having 1 signature after the tx was executed in safe 2

## Open issues

~The UI currently doesn't handle the response from mobile and is "stuck" in the modal flow. The new tx appears correctly in the queue of the safe that is connected but `sdk.approveTransactionHash` and `sdk.executeTransaction` are not resolving when the tx is confirmed on mobile.~
This has been solved by using an [unchecked signer](https://docs.ethers.io/v5/api/providers/jsonrpc-provider/#UncheckedJsonRpcSigner) which is similar to web3js `PromiEvent.once('transactionHash')`
